sklearn/utils/tests/test_indexing.py

```diff
@@ -495,6 +496,44 @@ def test_resample():
     assert len(resample([1, 2], n_samples=5)) == 5
 
 
+def test_resample_weighted():
+    # Check that sampling with replacement with weights yields the
+    # samples from the same distribution as sampling uniformly with
+    # repeated data points.
+    data = np.array([-1, 0, 1])
+    sample_weight = np.asarray([0, 100, 1])
+
+    mean_repeated = []
+    mean_reweighted = []
+
+    for seed in range(100):
+        mean_repeated.append(
+            resample(
+                data.repeat(sample_weight),
+                replace=True,
+                random_state=seed,
+                n_samples=data.shape[0],
+            ).mean()
+        )
+        mean_reweighted.append(
+            resample(
+                data,
+                sample_weight=sample_weight,
+                replace=True,
+                random_state=seed,
+                n_samples=data.shape[0],
+            ).mean()
+        )
+        # Should never be negative because -1 has a 0 weight.
```

Comment:
https://github.com/scikit-learn/scikit-learn/pull/29907/files#r1817718787 was marked resolved, but only half of the change was included.

```suggestion
        # Should never be negative because -1 has a 0 weight.
        assert mean_reweighted >= 0
```